src/Network.cpp

```diff
@@ -406,10 +406,13 @@ Network::Netresult Network::get_scored_moves_internal(
     assert(rotation >= 0 && rotation <= 7);
     constexpr int width = 19;
     constexpr int height = 19;
-    assert(INPUT_CHANNELS == planes.size());
-    std::vector<net_t> input_data;
-    auto convolve_channels = conv_pol_w.size() / conv_pol_b.size();
-    std::vector<net_t> output_data(convolve_channels * width * height);
+    constexpr int input_channels = INPUT_CHANNELS;
+    assert(input_channels == planes.size());
+    const auto input_size = input_channels * width * height;
+    const auto output_channels = int(conv_pol_w.size() / conv_pol_b.size());
```

Comment:
Don't need the cast here. (It's counterproductive: wouldn't you rather end up with a size_t?)

Reply:
I don't write a lot of c++, I'm sorry that you keep having to walk me (and others) though c++ basics and your preferred style.